### ISSUE TITLE
bug:  static calls fail on precompiles

### ISSUE THREAD

**Issue by neverDefined** [berachain/polaris#447]
```sh
warning[2018]: Warning: Function state mutability can be restricted to view
  --> src/examples/Deploy.s.sol:41:5:
   |
41 |     function run() public {
   |     ^ (Relevant source part starts here and spans across multiple lines).



Traces:
  [150016] → new Test@0x5b73C5498c1E3b4dbA84de0F1833c4a029d90519
    └─ ← 528 bytes of code

  [24022] Test::run()
    ├─ [0] 0xd9A998CaC66092748FfEc7cFBD155Aae1737C2fF::getActiveValidators() [staticcall]
    │   └─ ← ()
    └─ ← "EvmError: Revert"


Error:
Script failed.
```

**Comment by itsdevbear** [comment]
@calbera this would be the forge bug right? Gonna post an issue in their repo.

**Comment by calbera** [comment]
Yeah most likely. might be worth reproducing the error case on our latest code. Basically,  when there’s no return value, address(..).call and .staticcall work as expected, but calling the function directly fails. As you were saying, probably a decoding issue

**Comment by itsdevbear** [comment]
Yeah I have a feeling it's just a nil vs 0 error somewhere and it'll be fixed. 

The more I think about it, it shouldn't have to return anything to not error. Current always return bool definitely feels like a patch fix. As with the hacky, gas thing.

**Comment by calbera** [comment]
Probably a good idea to post it in the foundry repo for any suggestions from their team

**Comment by itsdevbear** [comment]
For sure, this seems to be an us issue tho. 

The foundry issue was because it's simulating on their evm and the precompile doesn't exist. 

This issue is just we are returning bad data from the precompile then.

**Comment by calbera** [comment]
We need to verify what bytes exactly is supposed to be returned when a function has no return values

**Comment by itsdevbear** [comment]
Yuh basically

**Comment by calbera** [comment]
Although this may be the reason even every ERC20 function returns something

**Comment by itsdevbear** [comment]
Doubtful. There are plenty of smart contacts that use functions with no return values.

**Comment by itsdevbear** [comment]
will be closed by #477
